src/main/java/org/elasticsearch/cluster/metadata/MappingMetaData.java

```diff
@@ -354,7 +354,7 @@ private void initMappers(Map<String, Object> withoutType) {
                     path = fieldNode.toString();
                 } else if (fieldName.equals("format")) {
                     format = fieldNode.toString();
-                } else if (fieldName.equals("default")) {
+                } else if (fieldName.equals("default") && fieldNode != null) {
```

Comment:
Valid point. Though I think null values might be rejected at mapping creation time for other values.
But as it's a safer option, I'll modify the PR.